docs/api.md

```diff
@@ -3744,8 +3744,38 @@ Inserts the instance to a different location. If the instance is currently in th
 
 **Examples**
 
+<div data-playground="N4IgFiBcoE5SBTAJgcwSAvgGhAZ3gEoCGAxgC4CWAbggHQkD2AtgA4MB2C7ZutR-AAgC8A4uWp0EADzJckACmAAddkrKzWAGyKzIApSAA8AIwCu6jgD5gwAJ4UEmpBgyGA9GYvtLBlRgCUKpwA7qKklDSKKmqOegDkAMQM5izmcVjR6ghaOgh6AAaZaoZgAMw+qmRqxYxICAIk2ri4QgbGZKoglrKaCGwwZO61CJYCnmQcAsEUmpoCZDBE7Lj9ZPNg9YbDDU0tbR0GlhuzDEMMdaPjk0VVZABmMMwCto6aDKFI1PMMAjDIAp8qDd3GUKtUqpVqpDBgBCAC0cIECORKNRaPRGIRSLhYNuJnME3YAg4cMaFBIAGtWiAmAwaIcen0GAN3FdvMC2cT2KTNOSqQY-rgJn9DoLhQhWQSrDdgYCBELbL1qcZmXUYHoAEwsKTyhi8pDPV7vADcAiYFG5GwoKDAZD0AFYAAzaw4ywb8IhHI3uD244puQF+wZyihIal3BgMAzysiKhDK1UIdUCACM2t1+t+yFN5stCGttodzqkruh7kDmXyWH0lQ4f3Yar0d1M7HEHHk-mU0LIwR+IjIYAovDuFqQAGFmGxONx5HEPXF-MabmQiDA0Gt+4Ph6PZwkIwwF0voczrQAZBgkHQUSb93u0FdrhAQ8EDoe0dtd8FqWk0JsttvsB2n63C+d4WrgSZkPID7roubrYG6YrMnkAjNq2lDtp2bpqD2DC0OBkHyCeKDnpeGHsHB0JqBgNwBEe1F+P4IA4GQ8CGCGYYGMkZCpFUXTltQljMSAJDwLQ7REsBjBvMmwSDrIpqZMYpAUigjwtkgpJ6syehqUQtiKZULBEEgnzsCgeilNqhlqCqMBqnCiyfKYuCWdZSmJsmaY6rgeqhmM2iUkeNHsJgQA"></div>
+
 ```js
-// TODO
+Ractive.components.aaa = Ractive.extend({
+	template: "<button>{{yield}}</button>"
+})
+Ractive({
+    ...
+	template: `
+		<button on-click="move">teleport</button>
+		<button on-click="restore">restore</button>
+
+		<div style="border: 2px solid yellow; min-height: 50px">
+			<aaa>hello</aaa>
+		</div>
+		<div id="foo" style="border: 1px solid red; min-height: 50px">
+		</div>
+	`,
+	onrender: function(){
+		two = this.findComponent('aaa');
+		target = this.find('#foo');
+		origLocation = two.target
+		this.on({
+			move: function(){
+				two.insert(target);
+			},
+			restore: function(){
+				two.insert(origLocation);
+			}
+		});
+	}
```

Comment:
Suggesting the use of the init option `on` instead of `this.on`. Nothing different really, but the init option is a bit more concise, avoids the lifecycle function, and perf isn't a concern in the example to warrant caching. Also, ID'ing both container divs would make both containers more explicit.

```
Ractive({
  template: `
    <button on-click="move">teleport</button>
    <button on-click="restore">restore</button>

    <div id="container1" style="border: 2px solid yellow; min-height: 50px">
      <aaa>hello</aaa>
    </div>

    <div id="container2" style="border: 1px solid red; min-height: 50px">

    </div>
  `,
  on: {
    move(){
      const aaa = this.findComponent('aaa')
      const target = this.find('#container2')
      aaa.insert(target)
    },
    restore(){
      const aaa = this.findComponent('aaa')
      const target = this.find('#container1')
      aaa.insert(target)
    }
  }
})
```

Reply:
But they are not the same examples. Mine was showing how to move a node to a known location and restore it. Suggested one shows how to move a node a known location twice.

Reply:
I know, the `init options` may seem clearer but most of the time I find myself doing many things on many components and many nodes inside a component via many handlers. At this point, closure is the saver: 

```ls
oninit: ->
    foo = $ @find '.something'
    bar = new Actor!
    @on do 
        myHandlerOne: (ctx) -> 
            foo.css background: \yellow 
            bar.send 'hello.there', {well: \done}

        myHandlerTwo: (ctx) -> 
            foo.css background: \error
            bar.send 'hello.two', {well: \error}        
```

How would I be able to do that with init options? I think there is no other way. 

Also, using *init options* break the "Don't Repeat Yourself" paradigm.

Reply:
> But they are not the same examples. 

Look closer 😉 

> most of the time I find myself doing many things on many components and many nodes inside a component via many handlers
>
> Also, using init options break the "Don't Repeat Yourself" paradigm.

But that is not the case of the example. The example should be like "Hey, you can do this in 3-ish lines" and not "Hey, I can do this faster but that's probably not why you're here.". 😁 

Lots of lib docs are written in a way that there's too many unnecessary asides (like adding unrelated APIs, written in a special manner, etc.), making the reader more confused than when they came - something I'd like to avoid.

Reply:
Yeah, looking closer, but no, they are not the same :) 

On the other hand, you are right about that it's more important to keep an example as simple as possible. So whether or not they are the same examples, I'm simplifying it per your request.